File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/SelfOnly.java
package com.anathema_roguelike.entities.characters.perks.actions.targetingstrategies.constraints;

import com.anathema_roguelike.entities.characters.Character;

public class SelfOnly implements TargetConstraint<Character, Character> {

	@Override
	public boolean apply(Character target, Character character) {
		return character == target;
	}
}

File: src/com/anathema_roguelike/entities/characters/perks/actions/targetingstrategies/constraints/SelfOnly.scala


package com.anathema_roguelike
package entities.characters.perks.actions.targetingstrategies.constraints
import com.anathema_roguelike.entities.characters.Character

class SelfOnly extends TargetConstraint[Character, Character] {
  override def apply(target: Character, character: Character): Boolean = {
    character == target
  }
}
